pinot-core/src/main/java/org/apache/pinot/core/plan/AggregationPlanNode.java

```diff
@@ -64,6 +64,10 @@
           DISTINCTCOUNTHLLMV, DISTINCTCOUNTRAWHLL, DISTINCTCOUNTRAWHLLMV, SEGMENTPARTITIONEDDISTINCTCOUNT,
           DISTINCTCOUNTSMARTHLL);
 
+  // DISTINCTCOUNT excluded because consuming segment metadata contains unknown cardinality when there is no dictionary
```

Comment:
wondering if we can somehow add to the logic of `isFitForMetadataBasePlan` checker for distinct count. basically one additional check to see if it is dictionary encoded?
we already did additional check for COUNT anyway. good to add DISTINCTCOUNT too?

Reply:
It would be handled by the dictionary based aggregator then. I can see an argument to merge those nodes, but needing to check the dictionary exists is a good reason not to.

Reply:
I just merged them in the end